common/client/node.go

```diff
@@ -224,53 +224,81 @@ func (n *node[CHAIN_ID, HEAD, RPC]) start(startCtx context.Context) {
-		n.declareUnreachable()
-		return
-	}
-
-	n.declareAlive()
+	state := n.verifyConn(startCtx, n.lfcLog)
+	n.declareState(state)
 }
 
-// verify checks that all connections to eth nodes match the given chain ID
+// verifyChainID checks that connection to the node matches the given chain ID
 // Not thread-safe
-// Pure verify: does not mutate node "state" field.
-func (n *node[CHAIN_ID, HEAD, RPC]) verify(callerCtx context.Context) (err error) {
+// Pure verifyChainID: does not mutate node "state" field.
+func (n *node[CHAIN_ID, HEAD, RPC]) verifyChainID(callerCtx context.Context, lggr logger.Logger) nodeState {
 	promPoolRPCNodeVerifies.WithLabelValues(n.chainFamily, n.chainID.String(), n.name).Inc()
 	promFailed := func() {
 		promPoolRPCNodeVerifiesFailed.WithLabelValues(n.chainFamily, n.chainID.String(), n.name).Inc()
 	}
 
 	st := n.State()
 	switch st {
-	case nodeStateDialed, nodeStateOutOfSync, nodeStateInvalidChainID:
+	case nodeStateDialed, nodeStateOutOfSync, nodeStateInvalidChainID, nodeStateSyncing:
 	default:
 		panic(fmt.Sprintf("cannot verify node in state %v", st))
 	}
 
 	var chainID CHAIN_ID
+	var err error
 	if chainID, err = n.rpc.ChainID(callerCtx); err != nil {
 		promFailed()
-		return fmt.Errorf("failed to verify chain ID for node %s: %w", n.name, err)
+		lggr.Errorw("Failed to verify chain ID for node", "err", err, "nodeState", n.State())
+		return nodeStateUnreachable
 	} else if chainID.String() != n.chainID.String() {
 		promFailed()
-		return fmt.Errorf(
+		err = fmt.Errorf(
 			"rpc ChainID doesn't match local chain ID: RPC ID=%s, local ID=%s, node name=%s: %w",
 			chainID.String(),
 			n.chainID.String(),
 			n.name,
 			errInvalidChainID,
 		)
+		lggr.Errorw("Failed to verify RPC node; remote endpoint returned the wrong chain ID", "err", err, "nodeState", n.State())
+		return nodeStateInvalidChainID
 	}
 
 	promPoolRPCNodeVerifiesSuccess.WithLabelValues(n.chainFamily, n.chainID.String(), n.name).Inc()
 
-	return nil
+	return nodeStateAlive
+}
+
+// createVerifiedConn - establishes new connection with the RPC and verifies that it's valid: chainID matches, and it's not syncing.
+// Returns desired state if one of the verifications fails. Otherwise, returns nodeStateAlive.
+func (n *node[CHAIN_ID, HEAD, RPC]) createVerifiedConn(ctx context.Context, lggr logger.Logger) nodeState {
+	if err := n.rpc.Dial(ctx); err != nil {
+		n.lfcLog.Errorw("Dial failed: Node is unreachable", "err", err, "nodeState", n.State())
+		return nodeStateUnreachable
+	}
+
+	return n.verifyConn(ctx, lggr)
+}
+
+// verifyConn - verifies that current connection is valid: chainID matches, and it's not syncing.
+// Returns desired state if one of the verifications fails. Otherwise, returns nodeStateAlive.
+func (n *node[CHAIN_ID, HEAD, RPC]) verifyConn(ctx context.Context, lggr logger.Logger) nodeState {
+	state := n.verifyChainID(ctx, lggr)
+	if state != nodeStateAlive {
+		return state
+	}
+
+	isSyncing, err := n.rpc.IsSyncing(ctx)
```

Comment:
Can we put an additional config setting here?
NodeIsSyncingEnabled = false[default]

This will allow us to skip IsSyncing() calls for chains/clients that don't reliably implement this functionality, or have bugs around this behavior.